src/conf_mode/interfaces_wireguard.py

```diff
@@ -61,8 +65,15 @@ def get_config(config=None):
             if 'disable' not in peer_config and 'host_name' in peer_config:
                 wireguard['peers_need_resolve'].append(peer)
 
+    # Check if interface is used as source-interface on VXLAN interface
+    if 'deleted' not in wireguard:
+        tmp = is_source_interface(conf, ifname, 'vxlan')
+        if tmp:
+            set_dependents('vxlan', conf, tmp)
+
     return wireguard
 
+
 def verify(wireguard):
```

Comment:
It would make more sense to error out here if the wireguard interface is a source interface of a vxlan interface and is about to be deleted.